### PR TITLE
experiment: small DiT model

### PR DESCRIPTION
**Pull request by bghira**

### CODE REVIEW

**Comment by bghira** [comment]
cc @sayakpaul

File: helpers/caching/text_embeds.py [hunk 1]
```diff
@@ -840,6 +842,7 @@ def compute_embeddings_for_legacy_prompts(
                         if "deepfloyd" not in StateTracker.get_args().model_type:
                             # we have to store the attn mask with the embed for pixart.
                             # aura doesn't require it, but it's just easier to keep.
+                            # smoldit requires the attn mask at inference time 💪🏽
```
Comment (sayakpaul) [inline helpers/caching/text_embeds.py:845]: ❣️

**Review by sayakpaul — APPROVED** [review]
Thanks!

Would it have been a better idea to just depend on https://github.com/sayakpaul/smoldit? This way, it's easier to pull in the changes. No?

**Comment by bghira** [comment]
yeah eventually i'd like to move the dependency out of tree but for now it's quickest to do this so i can ship it around without another moving piece

**Pull request merged**